lib/wallet/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -348,7 +347,8 @@ mkTransaction era networkId keyF stakeCreds addrResolver ctx cs = do
                     mempty
                 Just action ->
                     let stakeXPub = toXPub $ fst stakeCreds
-                    in certificateFromDelegationAction era (Left stakeXPub) action
+                    in certificateFromDelegationAction
+                        @era (Left stakeXPub) action
```

Comment:
```suggestion
                    in certificateFromDelegationAction (Left stakeXPub) action
```

The explicit type application does not improve over the explicit value application `era` — in fact, it makes things worse by requiring a language extension. The goal of the `IsRecentEra` type class is to pass the `era` as an **implicit** argument, i.e. to not pass it explicitly.

There are two cases:
* Either the type application is actually superfluous, then it can be removed,
* or there is an issue with the type inference of the return type. Probably, in this case the issue is that `let` is monomorphic due to the `MonoLocalBinds` extension, which is implied by the `GADTs` extension. The correct solution to this is to add a type annotation `certs :: [Cardano.Certificate (Write.CardanoApiEra era)]`.

Reply:
This particular type application was redundant. I removed it in https://github.com/cardano-foundation/cardano-wallet/compare/642df6baeeaa123bfca4dd48c14e81b58c2278a1..55ac25413a88da79a89b5848950a3a575fd6fc8c.